src/zenml/orchestrators/step_launcher.py

```diff
@@ -226,20 +226,53 @@ def launch(self) -> None:
 
                 logger.info(f"Step `{self._step_name}` has started.")
                 if execution_needed:
-                    try:
-                        self._run_step(
-                            pipeline_run=pipeline_run,
-                            step_run=step_run_response,
-                        )
-                    except BaseException as e:  # noqa: E722
-                        logger.error(
-                            f"Failed to run step `{self._step_name}`."
-                        )
-                        logger.exception(e)
-                        publish_utils.publish_failed_step_run(
-                            step_run_response.id
-                        )
-                        raise
+                    retries = 0
+                    max_retries = (
+                        step_run_response.config.retry.max_retries
+                        if step_run_response.config.retry
+                        else 1
+                    )
+                    delay = (
+                        step_run_response.config.retry.delay
+                        if step_run_response.config.retry
+                        else 0
+                    )
+                    backoff = (
+                        step_run_response.config.retry.backoff
+                        if step_run_response.config.retry
+                        else 1
+                    )
+
+                    while retries < max_retries:
+                        try:
+                            self._run_step(
```

Comment:
With the current logic, this will run the failure hook every time a try runs. Is that indented? Just logically I would expect that failure hook to only run in case all retries failed, otherwise I get notified about steps that didn't even cause my pipeline to fail?